### PR TITLE
[Frontend_v2] Edit forum URL

### PR DESCRIPTION
**Pull request by Kajol-Kumari**
Fixes: #2900 [Frontend V2: Add capability to edit forum URL by the challenge hosts ]

- Discuss tab will now have a button with link to the forum

- Challenge host can now edit the forum URL.

GIF of Feature:
edit-forum-url

cc: @RishabhJain2018 @Sanji515 @lunayach @Shekharrajak

### CODE REVIEW

File: frontend_v2/src/app/components/challenge/challengediscuss/challengediscuss.component.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<div class="challenge-card">
+  <div class="ev-card-panel card-bt-margin">
+    <div class="ev-md-container ev-panel-title">
+      <div class="row row-lr-margin">
+        <div class="col-sm-12 col-xs-12 col-lr-pad">
+          <h5 class="fw-light">Discuss</h5>
+          <span class="fs-16">Here you can discuss anything related to the <b>{{challenge.title}}</b>.Get all the anwers
```
Comment (Sanji515) [inline frontend_v2/src/app/components/challenge/challengediscuss/challengediscuss.component.html:7]: Minor: Hey @Kajol-Kumari  can we give a space here after full-stop (`.`) before the text `Get all the anwers for your queries, discuss any new idea......`

File: frontend_v2/src/app/components/challenge/challengediscuss/challengediscuss.component.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            for your queries, discuss any new idea or get to know the host team and other participants at the discussion
+            pannel.</span> <br> <br>
+          <div class="row">
+            <div class="col-md-2 col-sm-2 col-xs-6 col-lr-pad forum-btn">
+              <a class="ev-btn btn-waves-effect grad-rec-btn waves-dark ev-btn-dark
```
Comment (Sanji515) [inline frontend_v2/src/app/components/challenge/challengediscuss/challengediscuss.component.html:12]: Suggestion: I think here rounded button with border-radius: 20px will look more better than rectangular one and also if we can reduce the padding for this particular button to `10px 30px 10px 30px`. WDYT @Kajol-Kumari

Comment (Kajol-Kumari) [reply frontend_v2/src/app/components/challenge/challengediscuss/challengediscuss.component.html:12]: yes, @Sanji515 It's looking better than the rectangular one.
imageedit_2_3092706366

I have incorporated this. Thanks for suggesting.

**Comment by RishabhJain2018** [comment]
@Sanji515 Can you please take a look at this and approve this so that I can merge it.

File: frontend_v2/src/app/components/challenge/challenge.component.ts [hunk 3]
```diff
@@ -51,6 +51,16 @@ export class ChallengeComponent implements OnInit {
    */
   isParticipated = false;
 
+  /**
+   * Is Forum enabled in Challenge
+   */
+  isForumEnabled;
```
Comment (lunayach) [inline frontend_v2/src/app/components/challenge/challenge.component.ts:57]: Either initialize them or at-least give them a type (same for the variable below).

File: frontend_v2/src/app/components/challenge/challengediscuss/challengediscuss.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ChallengediscussComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
```
Comment (lunayach) [inline frontend_v2/src/app/components/challenge/challengediscuss/challengediscuss.component.spec.ts:22]: #ToDo please add test cases for these components later.

File: frontend_v2/src/app/components/challenge/challengediscuss/challengediscuss.component.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      ).subscribe(
+        data => {
+          SELF.challenge.forum_url = data.forum_url;
+          SELF.globalService.showToast('success', 'The challenge forum url is successfully updated!', 5);
+        },
+        err => {
+          SELF.globalService.handleApiError(err, true);
+          SELF.globalService.showToast('error', err);
+        },
+        () => console.log('EDIT-CHALLENGE-FORUM-URL-FINISHED')
```
Comment (lunayach) [inline frontend_v2/src/app/components/challenge/challengediscuss/challengediscuss.component.ts:81]: Please remove this.

File: frontend_v2/src/app/components/challenge/challengediscuss/challengediscuss.component.html (hunk 1 above)
Comment (Sanji515) [reply frontend_v2/src/app/components/challenge/challengediscuss/challengediscuss.component.html]: Hey @Kajol-Kumari here we both missed this type error, please correct `anwers` to `answers`.

Comment (Kajol-Kumari) [reply frontend_v2/src/app/components/challenge/challengediscuss/challengediscuss.component.html]: Thanks @Sanji515 for pointing this!

**Pull request merged**